Client/Frontend/Settings/CustomSearchViewController.swift

```diff
@@ -42,60 +54,67 @@ class CustomSearchViewController: SettingsTableViewController {
         }
     }
 
-    var successCallback: (() -> Void)?
-
-    fileprivate func addSearchEngine(_ searchQuery: String, title: String) {
+    private func addSearchEngine(_ searchQuery: String, title: String) {
         spinnerView.startAnimating()
 
         let trimmedQuery = searchQuery.trimmingCharacters(in: .whitespacesAndNewlines)
         let trimmedTitle = title.trimmingCharacters(in: .whitespacesAndNewlines)
 
-        createEngine(forQuery: trimmedQuery, andName: trimmedTitle).uponQueue(.main) { result in
-            self.spinnerView.stopAnimating()
-            guard let engine = result.successValue else {
+        Task {
+            do {
+                let engine = try await createEngine(query: trimmedQuery, name: trimmedTitle)
+                self.spinnerView.stopAnimating()
+                self.profile.searchEngines.addSearchEngine(engine)
+
+                CATransaction.begin() // Use transaction to call callback after animation has been completed
+                CATransaction.setCompletionBlock(self.successCallback)
+                _ = self.navigationController?.popViewController(animated: true)
+                CATransaction.commit()
+            } catch {
+                self.spinnerView.stopAnimating()
                 let alert: UIAlertController
-                let error = result.failureValue as? CustomSearchError
+                let error = error as? CustomSearchError
 
                 alert = (error?.reason == .DuplicateEngine) ?
                     ThirdPartySearchAlerts.duplicateCustomEngine() : ThirdPartySearchAlerts.incorrectCustomEngineForm()
 
                 self.navigationItem.rightBarButtonItem?.isEnabled = true
                 self.present(alert, animated: true, completion: nil)
-                return
             }
-            self.profile.searchEngines.addSearchEngine(engine)
-
-            CATransaction.begin() // Use transaction to call callback after animation has been completed
-            CATransaction.setCompletionBlock(self.successCallback)
-            _ = self.navigationController?.popViewController(animated: true)
-            CATransaction.commit()
         }
     }
 
-    func createEngine(forQuery query: String, andName name: String) -> Deferred<Maybe<OpenSearchEngine>> {
-        let deferred = Deferred<Maybe<OpenSearchEngine>>()
+    func createEngine(query: String, name: String) async throws -> OpenSearchEngine {
         guard let template = getSearchTemplate(withString: query),
-            let url = URL(string: template.addingPercentEncoding(withAllowedCharacters: .urlFragmentAllowed)!), url.isWebPage() else {
-                deferred.fill(Maybe(failure: CustomSearchError(.FormInput)))
-                return deferred
+              let url = URL(string: template.addingPercentEncoding(withAllowedCharacters: .urlFragmentAllowed)!),
```

Comment:
Should we replace the force here with a unwrap? `withAllowedCharacters: .urlFragmentAllowed)!`